core/src/main/scala/kafka/server/ReplicaAlterLogDirsThread.scala

```diff
@@ -76,7 +76,8 @@ class ReplicaAlterLogDirsThread(name: String,
     var partitionData: Seq[(TopicPartition, FetchData)] = null
     val request = fetchRequest.build()
 
-    val (topicIds, topicNames) = replicaMgr.metadataCache.topicIdInfo()
+    val topicIds = request.data().topics().asScala.map { topic => (topic.topic(), topic.topicId()) }.toMap
```

Comment:
nit: We could remove some parenthesis here and we usually use `map` with parenthesis instead of the curly braces when inline: `map { topic => ... }` -> `map(topic=> ....)`.